src/wasm/extract.cpp

```diff
@@ -64,8 +70,12 @@ bool file_output::write(char* data, size_t n) {
 
   position_ += n;
   total_written_ += n;
-
-  if (is_complete()) {
+  if (extractor::get().get_settings().debug_messages_enabled) {
```

Comment:
It would be good to unify logging handling, looks like you could use the "wasm_debug" here.

Reply:
This was done on purpose to potentially save some execution time, as there are two consecutive log writes handled by a single _if_ statement, which would make two _ifs_ when used with a _wasm_debug_.
It's not obvious whether a simple _if_ like this is optimized when compiling to WASM, and the main goal of this commit was to expand on actual extract functionality while keeping some of the added debug as an extra and not to overhaul the whole debug system.
The existing debug calls in _extract.cpp_ weren't even operational before introducing _wasm_debug_, for an unknown reason.